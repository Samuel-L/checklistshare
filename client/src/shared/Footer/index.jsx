import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';
import { withStyles } from 'material-ui/styles';

const styles = theme => ({
  base: {
    display: 'flex',
    justifyContent: 'end',
    marginRight: '20px',
  },
  link: {
    margin: '20px',
    textDecoration: 'none',
    color: theme.palette.primary.A700,
  },
});

const Footer = props => (
  <div className={props.classes.base}>
    <Link className={props.classes.link} to="/contact">Contact</Link>
  </div>
);

Footer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(Footer);
