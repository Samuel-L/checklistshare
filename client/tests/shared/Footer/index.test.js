import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter } from 'react-router-dom';

import Footer from '../../../src/shared/Footer';

describe('shared: Footer', () => {
  const classes = { base: '', link: '' };
  const tree = renderer.create(<HashRouter><Footer classes={classes} /></HashRouter>).toJSON();

  it('matches snapshot', () => {
    expect(tree).toMatchSnapshot(); 
  });
});
