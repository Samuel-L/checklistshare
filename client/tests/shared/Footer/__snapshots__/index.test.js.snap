// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared: Footer matches snapshot 1`] = `
<div
  className="Footer-base-1"
>
  <a
    className="Footer-link-2"
    href="#/contact"
    onClick={[Function]}
  >
    Contact
  </a>
</div>
`;
